### PR TITLE
Engineer's helmet

### PR DESCRIPTION
**Pull request by PrPleGoo**
Refactored HandheldLightComponent a bit, allows other lights now.
Implemented "on"-state as an overlay.
Added Engineer's helmet.
Split EquippedPrefix on ItemComponent into itself and ClothingEquippedPrefix on ClothingComponent.
Added netcode to Item- and ClothingComponent, to sync the fields.

### CODE REVIEW

File: Content.Server/GameObjects/Components/Items/Storage/ItemComponent.cs [hunk 1]
```diff
@@ -3,13 +3,19 @@
...
 
 namespace Content.Server.GameObjects
 {
     public class ItemComponent : StoreableComponent, IAttackHand
     {
         public override string Name => "Item";
+        public override uint? NetID => ContentNetIDs.ITEM;
+        public override Type StateType => typeof(ItemComponentState);
 
+        public string EquippedPrefix { get; set; }
```
Comment (PJB3005) [inline Content.Server/GameObjects/Components/Items/Storage/ItemComponent.cs]: Make this call `Dirty()` in the setter.

File: Content.Shared/GameObjects/Components/Items/ClothingComponentState.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using SS14.Shared.GameObjects;
+using SS14.Shared.Serialization;
+using System;
+
+namespace Content.Shared.GameObjects.Components.Items
+{
+    [Serializable, NetSerializable]
+    public class ClothingComponentState : ComponentState
```
Comment (PJB3005) [inline Content.Shared/GameObjects/Components/Items/ClothingComponentState.cs]: Make this a subclass of `ItemComponentState` so the item equipped prefix gets carried through too.

File: Content.Server/GameObjects/Components/Items/Clothing/ClothingComponent.cs [hunk 3]
```diff
@@ -8,10 +11,15 @@ namespace Content.Server.GameObjects
...
         public override string Name => "Clothing";
+        public override uint? NetID => ContentNetIDs.CLOTHING;
+        public override Type StateType => typeof(ClothingComponentState);
+
         public SlotFlags SlotFlags = SlotFlags.PREVENTEQUIP; //Different from None, NONE allows equips if no slot flags are required
 
         private int _heatResistance;
+        
         public int HeatResistance => _heatResistance;
+        public string ClothingEquippedPrefix { get; set; }
```
Comment (PJB3005) [inline Content.Server/GameObjects/Components/Items/Clothing/ClothingComponent.cs]: Call `Dirty()` on setter here.

File: Content.Shared/GameObjects/Components/Items/ClothingComponentState.cs (hunk 2 above)
Comment (PrPleGoo) [reply Content.Shared/GameObjects/Components/Items/ClothingComponentState.cs]: No, then the call to the base constructor gets jangled up by trying to throw it into the constructor for ItemComponent.

File: Content.Server/GameObjects/Components/Items/Storage/ItemComponent.cs (hunk 1 above)
Comment (PrPleGoo) [reply Content.Server/GameObjects/Components/Items/Storage/ItemComponent.cs]: fixed

File: Content.Server/GameObjects/Components/Items/Clothing/ClothingComponent.cs (hunk 3 above)
Comment (PrPleGoo) [reply Content.Server/GameObjects/Components/Items/Clothing/ClothingComponent.cs]: fixed

File: Content.Shared/GameObjects/Components/Items/ClothingComponentState.cs (hunk 2 above)
Comment (PJB3005) [reply Content.Shared/GameObjects/Components/Items/ClothingComponentState.cs]: Then make it take two params, one for the item equip prefix and one for the clothing equip prefix.

Either way the clothing component state *replaces* the item component state, so without doing this you won't be able to replicate the item equip prefix anymore for clothing items.

Comment (PrPleGoo) [reply Content.Shared/GameObjects/Components/Items/ClothingComponentState.cs]: Duplicated the field from ItemComponentState into ClothingComponentState.

**Comment by PrPleGoo** [comment]
Added pass-through constructor to ItemComponentState for inherritance by ClothingComponentState.

**Comment by PrPleGoo** [comment]
> You're no longer setting equipped prefix on the item component state.

I'm not sure if this is really a better solution.

**Pull request merged**